src/quo2/components/wallet/wallet_overview/view.cljs

```diff
@@ -0,0 +1,127 @@
+(ns quo2.components.wallet.wallet-overview.view
+  (:require [quo2.components.icon :as icons]
+            [quo2.components.markdown.text :as quo2]
+            [quo2.foundations.colors :as colors]
+            [react-native.core :as rn]
+            [utils.i18n :as i18n]
+            [quo2.components.wallet.wallet-overview.style :as style]
+            [quo2.theme :as quo.theme]))
+
+(def time-frames
+  {:one-week     (i18n/label :t/one-week-int)
+   :one-month    (i18n/label :t/one-month-int)
+   :three-months (i18n/label :t/three-months-int)
+   :one-year     (i18n/label :t/one-year)
+   :all-time     (i18n/label :t/all-time)})
+
+;; temporary placeholder for network dropdown component
+(defn- network-dropdown
+  []
+  [:<>
+   [rn/view
+    {:style {:border-color  colors/neutral-50
+             :border-width  1
+             :border-radius 10
+             :width         68
+             :height        32}}]])
+
+(defn- view-internal
+  [{:keys [state time-frame metrics balance date begin-date end-date
+           currency-change percentage-change theme]}]
+  (let [color-metrics (if (= metrics :positive)
+                        colors/success-50
+                        colors/danger-50)]
+    [rn/view {:style style/container-info}
+     [:<>
+      [rn/view {:style {:flex 1}}
+       [rn/view
+        {:style {:flex-direction  :row
+                 :flex            1
+                 :justify-content :space-between
+                 :align-items     :center}}
+
+        (if (= state :loading)
+          [rn/view {:style (style/loading-bar 201 20 0 theme)}]
+          [quo2/text
+           {:weight :semi-bold
+            :size   :heading-1
+            :style  {:color (colors/theme-colors colors/neutral-100 colors/white theme)}}
+           balance])
+        [network-dropdown]]
+
+       [rn/view {:style style/container-metrics}
+
+        (when (= state :loading)
+          [rn/view
+           {:style {:flex-direction :row
+                    :align-items    :center}}
+           [rn/view {:style (style/loading-bar 32 10 8 theme)}]
+           [rn/view {:style (style/loading-bar 32 10 4 theme)}]
+           [rn/view {:style (style/loading-bar 62 10 4 theme)}]
+           [rn/view {:style (style/loading-bar 10 10 0 theme)}]])
```

Comment:
Can we extract this into a separate function?
I think it'll shorten this function's height, which make it more readable.

Reply:
yes, done